tests/unit_tests/restruct_poc/test_controller.cpp

```diff
@@ -90,4 +91,74 @@ TEST(ControllerTest, DisconController) {
     EXPECT_FLOAT_EQ(avrSWAP[47], 0.);          // DemandedNacelleYawRate
 }
 
+TEST(ControllerTest, TurbineController) {
+    // Get a handle to the controller function via the TurbineController class and use it to
+    // calculate the controller outputs
+    std::string shared_lib_path = "./DISCON.dll";
+    std::string controller_function_name = "DISCON";
+
+    util::TurbineController controller(shared_lib_path, controller_function_name);
+
+    controller.io->status = 0.;
+    controller.io->time = 0.;
+    controller.io->pitch_blade1 = 0.;
+    controller.io->pitch_angle_actuator_req = 0.;
+    controller.io->generator_speed = 122.909576;
+    controller.io->horizontal_wind_speed = 11.9900799;
+    controller.io->pitch_blade2 = 0.;
+    controller.io->pitch_blade3 = 0.;
+    controller.io->generator_contactor_status = 1.;
+    controller.io->shaft_brake_status = 0.;
+    controller.io->demanded_yaw_actuator_torque = 0.;
+    controller.io->pitch_command_1 = 0.;
+    controller.io->pitch_command_2 = 0.;
+    controller.io->pitch_command_3 = 0.;
+    controller.io->pitch_command_collective = 0.;
+    controller.io->demanded_pitch_rate = 0.;
+    controller.io->demanded_generator_torque = 0.;
+    controller.io->demanded_nacelle_yaw_rate = 0.;
+    controller.io->message_array_size = 3.;
+    controller.io->infile_array_size = 82.;
+    controller.io->outname_array_size = 96.;
+    controller.io->pitch_override = 0.;
+    controller.io->torque_override = 0.;
+    controller.io->n_blades = 3.;
+    controller.io->n_log_variables = 0.;
+    controller.io->generator_startup_resistance = 0.;
+    controller.io->loads_request = 0.;
+    controller.io->variable_slip_status = 0.;
+    controller.io->variable_slip_demand = 0.;
+
+    controller.CallController();
+
+    EXPECT_FLOAT_EQ(controller.io->generator_contactor_status, 1.);    // GeneratorContactorStatus
+    EXPECT_FLOAT_EQ(controller.io->shaft_brake_status, 0.);            // ShaftBrakeStatus
+    EXPECT_FLOAT_EQ(controller.io->demanded_yaw_actuator_torque, 0.);  // DemandedYawActuatorTorque
+    EXPECT_FLOAT_EQ(controller.io->pitch_command_collective, 0.);      // PitchComCol
+    EXPECT_FLOAT_EQ(
+        controller.io->demanded_generator_torque, 43093.5508
+    );                                                              // DemandedGeneratorTorque
+    EXPECT_FLOAT_EQ(controller.io->demanded_nacelle_yaw_rate, 0.);  // DemandedNacelleYawRate
```

Comment:
Is there any need to check the ones that we just defined and we don't expect to change in the CallController call (i.e. demanded_yaw_actuator_torque)?

For the ones like demanded_generator_torque, which do change but aren't explicitly set, you should also check their values before the CallControler call, too.

Reply:
Working on it